### PR TITLE
ci: document the purpose of some non-obvious jobs

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -100,7 +105,7 @@ jobs:
       - name: Cache dependencies
         run: ./ci/before_cache.sh
 
-  unixish:
+  posix:
```
Comment (zeertzjq) [inline .github/workflows/ci.yml:108]: What's wrong with "unixish"?

Comment (dundargoc) [reply .github/workflows/ci.yml:108]: I recall someone mentioning that posix was the more appropriate term. It's not an important distinction to me, I'll revert it.

Comment (clason) [reply .github/workflows/ci.yml]: Yep; "Posix" is a well-defined term, while "unixish" is not.

Comment (zeertzjq) [reply .github/workflows/ci.yml]: Ah, I first thought "posix" is too general, but it turns out to be the other way around.

Comment (dundargoc) [reply .github/workflows/ci.yml]: OK, re-adding the posix then :)

**Pull request merged**